### ISSUE TITLE
Better initial errors for poorly formed YAML/JSON

### ISSUE THREAD

**Issue by cmmeyer** [awslabs/cfn-python-lint#417]
*cfn-lint version: (`cfn-lint --version`)*
0.8.2

*Description of issue.*
Some of the entry-point errors for really broken templates are very cryptic.  We should identify situations where it makes more sense to drop back to general YAML/JSON linting or a generic "JSON/YAML is unparseable" error.

Our error:
`E0000 Null value at line 10 column 37`

CloudFormation `aws cloudformation validate-template` error:
`An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: [/Resources/PublicSubnetRouteTableAssociation1] resource definition is malformed`

We should be at least as good.

Template:

```yaml
Parameters:
  PublicRouteTable:
    Type: String
    Default: rt-123
  PublicSubnet01:
    Type: String
    Default: s-123456
Conditions:
  CreateAuxiliaryNetworks: !Equals [ !Ref PulicSubnet01, prod ]
Resources:
  PublicSubnetRouteTableAssociation1:
  Type: AWS::EC2::SubnetRouteTableAssociation
  Properties:
    RouteTableId: {Ref: PublicRouteTable}
    SubnetId: {Ref: PublicSubnet01}

  AuxiliaryPublicSubnetRouteTableAssociation1:
    Type: AWS::EC2::SubnetRouteTableAssociation
    Condition: CreateAuxiliaryNetworks
    Properties:
      RouteTableId: {Ref: PublicRouteTable}
      SubnetId: {Ref: PublicSubnet01}
```

**Comment by kddejong** [comment]
So you want the location in there as well (/Resources/PublicSubnetRouteTableAssociation1)?  Obviously we are providing the location based on column and line # but the message itself is a little lacking.  Thats cause we are detecting the null errors during the parse operations and not as a separate rule.

**Comment by rjlohan** [comment]
I'm not sure even the line # is correct here? It complains about line 10, but it's line 12-15 where my indenting got broken.

**Comment by kddejong** [comment]
The location matches for the `PublicSubnetRouteTableAssociation1` as it has a null value in this case.

**Comment by kddejong** [comment]
I think there is an opportunity here to change some things and make the messages better.   I'm just trying to make sure I understand the goal correctly.

#40 - is the change that handled null value errors for reference.  Not a lot there but the goal was to fail with some grace before we would run into issues like #403 when trying to process the rules.  I have a pull request there that could be used similarly.  There is probably a more strategic design here given our use case.

**Comment by rjlohan** [comment]
> The location matches for the `PublicSubnetRouteTableAssociation1` as it has a null value in this case.

Is the count 0-indexed? Because that resource is on L11 by my count? That's part of my initial confusion with this error message.

**Comment by kddejong** [comment]
I see what you are saying now.  It is zero based.  Which isn't ideal

**Comment by kddejong** [comment]
So I was looking at this again.  The generic error string we got is off by one but the location we print is actually correct.   `11:38`.  @rjlohan would it just work to correct the error message?
E0000 Null value at line 10 column 37
test1.yaml:11:38

**Comment by cmmeyer** [comment]
This is now waaaaaaay better:

```
cfn-lint whats-this-do.yaml
E0000 Null value at line 11 column 38
whats-this-do.yaml:11:38
```
